### DISCUSSION TITLE
fatal error: hb.h: No such file or directory

### DISCUSSION THREAD

**Discussion by og900aero** [bakkeby/st-flexipatch#155] [General]
I git clone the repo, changed patches.def.h, and sudo make clean install, and I get error message:

```
sudo make clean install
rm -f st st.o x.o st-0.9.2.tar.gz
c99 -I/usr/X11R6/include  `pkg-config --cflags fontconfig`  `pkg-config --cflags freetype2`   -DVERSION=\"0.9.2\" -DICON=\"/usr/local/share/pixmaps/st.png\" -D_XOPEN_SOURCE=600  -O1 -c st.c
c99 -I/usr/X11R6/include  `pkg-config --cflags fontconfig`  `pkg-config --cflags freetype2`   -DVERSION=\"0.9.2\" -DICON=\"/usr/local/share/pixmaps/st.png\" -D_XOPEN_SOURCE=600  -O1 -c x.c
In file included from x.c:23:
hb.h:2:10: fatal error: hb.h: No such file or directory
    2 | #include <hb.h>
      |          ^~~~~~
compilation terminated.
make: *** [Makefile:19: x.o] Error 1
```
If I try oldest git cloned (0.9.1, Apr 23 maybe), then working fine.

**Comment by og900aero** [comment]
I installed all harfbuzz package on Debian 12, but I get error message again.

**Reply by bakkeby** [reply]
Some patches require additional libraries. This is usually mentioned in the patch toggle comment.

https://github.com/bakkeby/st-flexipatch/blob/c4c5113fbdfb97363c510abb89a8da7f4fb138b2/patches.def.h#L228-L236

Unfortunately there is no straightforward way to include those libraries in the Makefile based on what you have enabled in your patches.h file, so you will need to manually add these in the config.mk file.

In these situations it should normally just be a matter of uncommenting a few lines. 

https://github.com/bakkeby/st-flexipatch/blob/c4c5113fbdfb97363c510abb89a8da7f4fb138b2/config.mk#L23-L27
